libraries/Microsoft.Bot.Builder.Ai/TranslationMiddleware.cs

```diff
@@ -118,20 +105,31 @@ public TranslationMiddleware(HttpClient httpClient, string[] nativeLanguages, st
                 string[] lines = text.Split(new string[] { "\r\n", "\n" }, StringSplitOptions.None);
                 var translateResult = await this.translator.TranslateArray(lines, sourceLanguage, targetLanguage).ConfigureAwait(false);
                 text = String.Join("\n", translateResult);
-
                 // restore mentions and urls 
                 i = 0;
-                //foreach (var mention in message.Mentions)
-                //{
-                //    text = text.Replace($"{{{i++}}}", mention.Text);
-                //}
                 foreach (var url in urls)
                 {
                     text = text.Replace($"{{{i++}}}", url);
                 }
 
                 message.Text = text;
             }
-        }        
+        }
+
+        public async Task SendActivity(IBotContext context, IList<IActivity> activities, MiddlewareSet.NextDelegate next)
+        {
+            foreach (var activity in activities)
+            {
+                IMessageActivity message = context.Request.AsMessageActivity();
+                if (!String.IsNullOrWhiteSpace(message?.Text))
+                {
+                    // translate to userslanguage
+                    var translationContext = ((BotContext)context)["Translation"] as TranslationContext;
+                    if (translationContext.SourceLanguage != translationContext.TargetLanguage && message.Text != translationContext.SourceText)
```

Comment:
>translationContext [](start = 24, length = 18)

this looks like it can be null if you didn't translate the incoming activity.

Reply:
I added a checking condition for null